### DISCUSSION TITLE
Improve error message when returning props.data from field hook

### DISCUSSION THREAD

**Discussion by AlessioGr** [payloadcms/payload#6750] [Feature Requests & Ideas]
### To Reproduce

```ts
{
      name: 'text',
      type: 'text',
      hooks: {
        afterRead: [(props) => {
          return props.data;
        }],
      },
    },
```
    
 Will throw the following error:
 [12:24:10] ERROR (payload): RangeError: Maximum call stack size exceeded
    at [Symbol.hasInstance] (<anonymous>)
    at deepCopyObject (/Users/alessio/Documents/GitHub/payload/dist/utilities/deepCopyObject.js:4:17)
    at /Users/alessio/Documents/GitHub/payload/dist/utilities/deepCopyObject.js:14:74
    at Array.forEach (<anonymous>)
    at deepCopyObject (/Users/alessio/Documents/GitHub/payload/dist/utilities/deepCopyObject.js:11:27)
    at /Users/alessio/Documents/GitHub/payload/dist/utilities/deepCopyObject.js:14:74
    at Array.forEach (<anonymous>)
    at deepCopyObject (/Users/alessio/Documents/GitHub/payload/dist/utilities/deepCopyObject.js:11:27)
    at /Users/alessio/Documents/GitHub/payload/dist/utilities/deepCopyObject.js:14:74
    at Array.forEach (<anonymous>)

### Describe the Bug

Optimally, it should be typed in a way where you can only return props.value / the correct value, and not props.data.

In any case, the error message should be improved. This is something which could happen by accident, as it's totally okay to return `props.data` from a collection hook, but field hooks need to return `props.value` instead. (And yes, this just happened to brainless me. I can imagine I wouldn't be the only one)

### Payload Version

1.9.1

### Related PR
#2851

**Comment by carstenblt** [comment]
Thank god this is the Google result.

**Comment by emwadde** [comment]
During development i had a hook that did nothing and an error like this was triggered. Thank God this was the first Google result.
Here is the hook I had
```
hooks: {
  beforeValidate: [
    ({data}) => {
      
      return data;
    }
  ]
}
```
And here is the error that was thrown
`ERROR (payload): RangeError: Maximum call stack size exceeded
    at deepCopyObject`

**Reply by emwadde** [reply]
Of cause I made a mistake, but the error message wasn't very helpful
